workflows/detaxizer.nf

```diff
@@ -189,31 +183,30 @@ workflow DETAXIZER {
     //
     // MODULE: Prepare Kraken2 Database
     //
-    ch_kraken2_db_with_meta = Channel.fromPath(params.kraken2db).map {
+    ch_kraken2_db = Channel.fromPath(params.kraken2db).map {
             item -> [['id': "kraken2_db"], item]
-            }
+        }
     KRAKEN2PREPARATION (
-        ch_kraken2_db_with_meta
+        ch_kraken2_db
     )
     ch_versions = ch_versions.mix(KRAKEN2PREPARATION.out.versions)
 
     //
     // MODULE: Run Kraken2
     //
-    ch_kraken2_db = FASTP.out.reads.combine(KRAKEN2PREPARATION.out.db).map{ it -> [it[3]]}
+    ch_kraken2_db = FASTP.out.reads.combine(KRAKEN2PREPARATION.out.db).map{ it -> [it[3]] }.first()
```

Comment:
```suggestion
    ch_kraken2_db =KRAKEN2PREPARATION.out.db.first()
```

I think you should achieve the same using this. The .first() will generate a value channel from the queue channel.

Reply:
indeed, that is what I meant.

Reply:
Recognized that when I do this, only one process of kraken2 is executed ... my initial work-around fixes this. Not sure why it is showing that behavior as a value channel should be consumed as often as needed when only a single value is inside of it ... the map is needed to drop meta from the db preparation as it is not used in the kraken2 module